diff --git a/nrsc5.py b/nrsc5.py
--- a/nrsc5.py
+++ b/nrsc5.py
@@ -510,6 +510,13 @@ class NRSC5:
     def pipe_samples_cs16(self, samples):
         if len(samples) % 4 != 0:
             raise NRSC5Error("len(samples) must be a multiple of 4.")
         result = NRSC5.libnrsc5.nrsc5_pipe_samples_cs16(self.radio, samples, len(samples) // 2)
         if result != 0:
             raise NRSC5Error("Failed to pipe samples.")
+
+    def pipe_samples_cf32(self, samples):
+        if len(samples) % 8 != 0:
+            raise NRSC5Error("len(samples) must be a multiple of 8.")
+        result = NRSC5.libnrsc5.nrsc5_pipe_samples_cf32(self.radio, samples, len(samples) // 4)
+        if result != 0:
+            raise NRSC5Error("Failed to pipe samples.")
